### DISCUSSION TITLE
FAQ (PLEASE READ FIRST)

### DISCUSSION THREAD

**Discussion by rickstaa** [anuraghazra/github-readme-stats#1770] [General]
Here you can find frequently asked questions. Please check this list and the [already asked questions](https://github.com/anuraghazra/github-readme-stats/discussions/categories/q-a) before [posting your question](https://github.com/anuraghazra/github-readme-stats/discussions/new?category=q-a). For common error codes see [Common Error Codes](https://github.com/anuraghazra/github-readme-stats/discussions/1772).

- [General questions](#general-questions)
  - [My card displays an error](#my-card-displays-an-error)
  - [I could not figure out how to deploy on my own Vercel instance](#i-could-not-figure-out-how-to-deploy-on-my-own-vercel-instance)
  - [How to count private stats?](#how-to-count-private-stats)
  - [My card is not updating](#my-card-is-not-updating)
- [Stats card related questions](#stats-card-related-questions)
  - [Number of commits is incorrect](#number-of-commits-is-incorrect)
- [Language card related questions](#language-card-related-questions)
  - [How to hide Jupyter Notebook?](#how-to-hide-jupyter-notebook)
  - [Language card incorrect](#language-card-is-incorrect)

### My card displays an error <a href="#my-card-displays-an-error" id="my-card-displays-an-error"/>

Please check [common error codes](https://github.com/anuraghazra/github-readme-stats/discussions/1772) and [existing issues](https://github.com/anuraghazra/github-readme-stats/issues) before creating a [new issue](https://github.com/anuraghazra/github-readme-stats/issues/new/choose).

### I could not figure out how to deploy on my own Vercel instance <a href="#i-could-not-figure-out-how-to-deploy-on-my-own-vercel-instance" id="i-could-not-figure-out-how-to-deploy-on-my-own-vercel-instance"/>

Please check out [the docs](https://github.com/anuraghazra/github-readme-stats/#deploy-on-your-own-vercel-instance) or the accompanying [Youtube tutorial](https://www.youtube.com/watch?v=n6d4KHSKqGk&feature=youtu.be&t=107) by @codeSTACKr.

### How to count private stats? <a href="#how-to-count-private-stats" id="how-to-count-private-stats"/>

By default, the stats card shows contributions only from public repositories. To show your private contributions, you should [deploy your own instance](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own) using your own GitHub API token, allowing you to see your private data.

### My card is not updating <a href="#my-card-is-not-updating" id="my-card-is-not-updating"/>

We use a cache to lower the server load, so please be patient before opening an issue. See <https://github.com/anuraghazra/github-readme-stats#gradient-in-bg_color> for more information.

### Number of commits is incorrect <a href="#number-of-commits-is-incorrect" id="number-of-commits-is-incorrect"/>

As we directly use the data provided by Github's GraphQL API, the commits shown on the card are probably correct. Most of the time, people confuse contributions with commits (see #1679 for more information) or are unaware caching is used to reduce the server load (see https://github.com/anuraghazra/github-readme-stats#gradient-in-bg_color). You can check the data we receive by running the following syntax in the [GraphQL explorer](https://docs.github.com/en/graphql/overview/explorer):

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    name
    login
    contributionsCollection {
      totalCommitContributions
      restrictedContributionsCount
    }
  }
}
```
You can then use the following in the query variables field.

```graphql
{"login": "rickstaa"}
```

If you enabled the `include_all_commits` option, the total commits are fetched using Github's [REST API](https://docs.github.com/en/rest/guides/getting-started-with-the-rest-api) instead. You can check the total commits we receive from this API using the following syntax in the [reqbin API tool](https://reqbin.com/):

```md
https://api.github.com/search/commits?q=author:<USER_NAME>
```

Please be aware that this, by default, only fetches your public commits. You must include an Authorization token if you want to retrieve your private commits (see https://docs.github.com/en/rest/guides/getting-started-with-the-rest-api#authentication). If you still think the number of commits is incorrect, please open [an issue](https://github.com/anuraghazra/github-readme-stats/issues).

### How to hide Jupyter Notebook? <a href="#how-to-hide-jupyter-notebook" id="how-to-hide-jupyter-notebook"/>

You can use the `hide` parameter (i.e. `&hide=jupyter%20notebook`). See <https://github.com/anuraghazra/github-readme-stats#hiding-individual-stats> for more information.

### Language card is incorrect <a href="#language-card-is-incorrect" id="language-card-is-incorrect"/>

As we directly use the data provided by Github's GraphQL API, the results shown on the card are probably correct. Most of the time, people are unaware that caching is used to reduce the server load (see https://github.com/anuraghazra/github-readme-stats#gradient-in-bg_color), that data from forks are not included (see #1255) or that there exists an upstream issue with the GraphQL API that prevents us from getting more accurate language results (see https://github.com/anuraghazra/github-readme-stats/issues/1801#issuecomment-1176153879). You can check the data we receive by running the following syntax in the [GraphQL explorer](https://docs.github.com/en/graphql/overview/explorer):

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    # fetch only owner repos & not forks
    repositories(ownerAffiliations: OWNER, isFork: false, first: 100) {
      nodes {
        name
        languages(first: 10, orderBy: {field: SIZE, direction: DESC}) {
          edges {
            size
            node {
              color
              name
            }
          }
        }
      }
    }
  }
}
```

You can then use the following in the query variables field.

```graphql
{"login": "rickstaa"}
```

**Comment by rickstaa** [comment]
Please comment below if you think a commonly asked question is missing from the FAQ 👍🏻.

**Reply by rassellek** [reply]
The error "no description provided" appears in "GitHub Extra Pins" while the description is present in the repository itself.

https://github.com/rassellek

**Reply by rickstaa** [reply]
@rassellek That looks like a bug. Please open a bug report on https://github.com/anuraghazra/github-readme-stats/issues/new?assignees=&labels=bug&template=bug_report.yml. 👍🏻

**Comment by benpaddlejones** [comment]
Can this work for an organisation or just a user?
